### ISSUE TITLE
Payment failure doesn't allow us to retry

### ISSUE THREAD

**Issue by mmiszy** [saleor/react-storefront#562]
Report from user:
> Payment failure doesn't allow us to retry. What if I made a typo and wanted to re-enter the card number or want to try another card?
> Payment failures cause the checkout to no longer exist, changing the form causes errors, and reloading the page causes the cart to be empty.

Context: The problem occurs because we first create an order which makes the checkout token no longer valid... So when a payment fails, we cannot retry the process because checkout no longer exists and we don't store previously created orders. We need to change the architecture of how this works a little bit.
We should be able to fetch order by checkoutId.

**Comment by mmiszy** [comment]
Blocked by https://github.com/saleor/saleor/issues/11009
